### ISSUE TITLE
Add a Vue example

### ISSUE THREAD

**Issue by kaisalmen** [TypeFox/monaco-languageclient#416]
Should we add a Vue client to the examples?

It should only be integrated into this repository if it does not bring a ton of dependencies (like Angular) and fits the overall dev approach with vite. But with VueI am positive it will work. Otherwise, it could live in an external repo.

**Comment by RSS1102** [comment]
I think we can pay attention to [language-tools(volar)](https://github.com/vuejs/language-tools#packages).
I am also trying to add a Vue3 language on Monaco. I think this is helpful.

**Comment by kaisalmen** [comment]
Hi @RSS1102 can you be more specific?

**Comment by RSS1102** [comment]
> Hi @RSS1102 can you be more specific?

This is my personal understanding：
Each language has its own pattern, and Vue is no exception. In order to increase portability, such as compatibility with Vue systems such as vscode/vitepress, a separate warehouse has been developed for Vue.

1. this is [vue-language-configuration.json](https://github.com/vuejs/language-tools/blob/master/extensions/vscode/languages/vue-language-configuration.json),Here are the key strings for the Vue language.
2. [@volar/monaco](https://github.com/volarjs/volar.js/tree/master/packages/monaco#volarmonaco), is used to bridge the language capabilities implemented based on Volar.js to Monaco Editor.

https://github.com/Kingwl/monaco-volar ,
There is an example code here, but I don't think it is perfect, there are too many dependencies, and many features are outdated. I personally think that all you need is the [vue-language-configuration.json](https://github.com/vuejs/language-tools/blob/master/extensions/vscode/languages/vue-language-configuration.json) file,and the rest needs to be done by ourself.
I personally believe that these functions mainly support:
1. Vue language support.
2. Language highlighting
3. Vue worker.

---
But I don't have enough understanding to implement it yet. But I am still working hard.

**Comment by kaisalmen** [comment]
My understanding is that volar helps to ease building languageservers / languageclient. What we gain with `@codingame/monaco-vscode-api` is there is no more monaco-editor related fiddling needed as it behaves like it is running inside vscode. So [monaco-volar](https://github.com/Kingwl/monaco-volar) is not needed as you can just use your languageclient implementation. @CGNonofr anything to correct or add here?

**Comment by RSS1102** [comment]
Yes, so maybe it's just a need for [vue-language-configuration.json](https://github.com/vuejs/language-tools/blob/master/extensions/vscode/languages/vue-language-configuration.json),i think.

**Comment by kaisalmen** [comment]
Yes that and a textmate grammar. That is what we load for the Langium Statemachine DSL as well: https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/src/langium/statemachineClient.ts#L77-L78
(The references are used in the extension configuration (a few lines above).

**Comment by RSS1102** [comment]
Okay, thank you. I'll give it a try this weekend.

**Comment by kaisalmen** [comment]
@RSS1102 are you still interested in doing this?

**Comment by RSS1102** [comment]
@kaisalmen i am so sorry~ 

I was busy with other work that I forgot about this issue. It's been so long. Let me see if I can continue to support this issue~
I will give you a reply or PR in the next few days，If you need it urgently, you can give me up..

**Comment by kaisalmen** [comment]
Hi @RSS1102 there is no need for urgency. I will very like not get to it, soon. Any help is appreciated.

**Comment by RSS1102** [comment]
@kaisalmen  Hello, I'm very sorry that this matter has been delayed for so long. I decided to try to continue to complete this promise, but it has been too long. I seem to have forgotten what I should do. I'm not sure if I can still complete it...
🥹🥹🥹🥹
Can you give me a suggestion again so that I know how I should start?

**Comment by kaisalmen** [comment]
@RSS1102 we have workspace independent verification examples. You can use this as a good starting point:
https://github.com/TypeFox/monaco-languageclient/tree/main/verify/vite

There are multiple verification examples, but the vite one fits best to integrate vue, I think.

**Comment by kaisalmen** [comment]
We didn't accomplish this in the last four years, so I will close it. If there arises the need, we can re-consider it.
